client/rpmtrans.c

```diff
@@ -458,15 +450,8 @@ TDNFTransAddInstallPkg(
         BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
     }
 
-    //Check override, then repo config and launch
-    //gpg check if needed
-    dwError = TDNFGetGPGSignatureCheck(pTdnf, pszRepoName, &nGPGSigCheck, &pszUrlGPGKey);
+    dwError = TDNFGetGPGSignatureCheck(pTdnf, pszRepoName, &nGPGSigCheck, NULL);
```

Comment:
The first call wouldn't bail if there is no key configured in the repo. It's not required at that point because we may already have the key in the rpm keyring, and at this point we only need to know if need to check the signature (by checking `nogpgcheck` or `skipsignature`. Only if the required key is not in the keyring do we need it configured, in the second call.